packages/mollie-integration/src/mollie-payment-strategy.ts

```diff
@@ -69,23 +72,23 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
 
         each(controllers, (controller) => controller.remove());
 
-        const state = this._store.getState();
-        const storeConfig = state.config.getStoreConfig();
+        const storeConfig = this.paymentIntegrationService.getState().getStoreConfig();
 
         if (!storeConfig) {
             throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
         }
 
-        this._initializeOptions = mollie;
+        this.initializeOptions = mollie;
+
+        const paymentMethod = this.paymentIntegrationService
+            .getState()
+            .getPaymentMethodOrThrow(methodId, gatewayId);
 
-        const paymentMethods = state.paymentMethods;
-        const paymentMethod = paymentMethods.getPaymentMethodOrThrow(methodId, gatewayId);
         const {
             config: { merchantId, testMode },
         } = paymentMethod;
-        const { locale } = paymentMethod.initializationData;
 
-        this._locale = locale;
+        this.locale = this.paymentIntegrationService.getState().getLocale();
```

Comment:
Same here

Reply:
As far as I see, `locale` value was used from `paymentMethod.initializationData`. Did you have a chance to check the difference between `state.getLocale()` and `paymentMethod.initializationData.locale`?